matchms/importing/load_from_msp.py

```diff
@@ -0,0 +1,100 @@
+from typing import Generator
+from typing import List
+import numpy
+from ..Spectrum import Spectrum
+
+
+def parse_msp_file(filename: str) -> List[dict]:
+    """Read msp file and parse info in list of spectrum dictionaries."""
+
+    # List that will contain all the differente "spectrums"
+    spectrums = []
```

Comment:
Use generator instead of list return
```suggestion
def parse_msp_file(filename: str) -> Generator[dict, None, None]:
    """Read msp file and parse info in list of spectrum dictionaries."""
```